packages/rocketchat-file-upload/server/lib/FileUpload.js

```diff
@@ -1,43 +1,264 @@
-/* globals FileUpload:true */
+/* globals UploadFS */
+
+import fs from 'fs';
 import mime from 'mime-type/with-db';
+import Future from 'fibers/future';
+
+Object.assign(FileUpload, {
+	handlers: {},
+
+	configureUploadsStore(store, name, options) {
+		const type = name.split(':').pop();
+		const stores = UploadFS.getStores();
+		delete stores[name];
+
+		return new UploadFS.store[store](Object.assign({
+			name
+		}, options, FileUpload[`default${ type }`]()));
+	},
+
+	defaultUploads() {
+		return {
+			collection: RocketChat.models.Uploads.model,
+			filter: new UploadFS.Filter({
+				onCheck: FileUpload.validateFileUpload
+			}),
+			getPath(file) {
+				return `${ RocketChat.settings.get('uniqueID') }/uploads/${ file.rid }/${ file.userId }/${ file._id }`;
```

Comment:
`file._id` is `undefined`.. that means all files are being saved as `undefined` (and being all replaced)..